webpack/assets/javascripts/react_app/components/factCharts/index.js

```diff
@@ -39,6 +39,8 @@ class FactChart extends React.Component {
       modalToDisplay,
     } = this.props;
 
+    const hostsCount = factChart.hostsCount || 0;
```

Comment:
The counter still shows on the modal title (just an odd `( hosts)`) while loading or no data exists:
![counting](https://user-images.githubusercontent.com/11807069/46951341-5b464500-d090-11e8-960b-4fb0ffddcb47.png)

IMHO this is better:
![notitle](https://user-images.githubusercontent.com/11807069/46951437-a3656780-d090-11e8-9b6e-f153ddeaf215.png)